runtime/commonMain/src/kotlinx/serialization/SerialKinds.kt

```diff
@@ -259,3 +266,20 @@ public sealed class PolymorphicKind : SerialKind() {
     public object OPEN : PolymorphicKind()
 }
 
+@Deprecated(
+    level = DeprecationLevel.ERROR,
+    message = "This object is added only as temporary migration mechanism"
```

Comment:
`UnionKind` is not a temporary migration mechanism, it's literally a deprecated object that existed